packages/core/src/utilities.ts

```diff
@@ -32,3 +32,27 @@ export function extractStyles(style: StyleProp<any>) {
 
   return { viewStyles, textStyles };
 }
+
+/**
+ * Merges a style object on top of another style object. In React Native,
+ * keys with undefined values in a style object will still override styles
+ * that appear earlier in a sequence. This avoids that problem.
+ */
+export function applyStyles(
+  baseStyles: StyleProp<any>,
+  stylesToApply: StyleProp<any> | undefined
+) {
+  if (!stylesToApply) {
+    return;
+  }
+
+  const flattenedStyles = StyleSheet.flatten(baseStyles);
```

Comment:
will the base styles always be unflattened? for example, if i run stylesheet.flatten outside of this function?

do you think we should add a check to see if its already flat? maybe this part isn't possible, but i could have sworn styles returns a number, and once flattened, returns an object

Reply:
It seems that, at least on the TS side, `StyleSheet.flatten` expects an array of objects always ((docs)[https://reactnative.dev/docs/stylesheet#flatten]) - so maybe what I can do is enforce the type of `baseStyles` on this function to take an array of objects, so we know that the styles aren't already flattened.

Reply:
245a9a8